js/components/interface/query/customComponents/queryLinkComponent.js

```diff
@@ -0,0 +1,38 @@
+define(function (require) {
+
+  require("../query.less");
+  require("../react-simpletabs.less");
+
+  var React = require('react');
+  var GEPPETTO = require('geppetto');
+
+  class QueryLinkComponent extends React.Component {
+    constructor (props) {
+      super(props);
+    }
+
+    render () {
+    
+      var displayText = this.props.data;
+      var path = this.props.rowData.id;
+      var that = this;
+    
+      var action = function (e) {
+        e.preventDefault();
+        e.nativeEvent.stopImmediatePropagation();
+        var actionStr = that.props.metadata.actions;
+        actionStr = actionStr.replace(/\$entity\$/gi, path);
+        GEPPETTO.CommandController.execute(actionStr);
+        GEPPETTO.QueryBuilder.close();
```

Comment:
@ddelpiano commandController is ok but would be good to remove the GEPPETTO.QueryBuilder reference as part of this refactoring

Reply:
This has been removed from almost everywhere, the only point where I have this at the moment it's the QueryBuilderConfiguration, in this [point](https://github.com/VirtualFlyBrain/geppetto-vfb/blob/vfb_geppetto_application/components/configuration/queryBuilderConfiguration.js#L107) , this will be removed as part of [this](https://github.com/openworm/geppetto-client/issues/44) if you agree as part of INFRA if you agree.